src/convertToObject.js

```diff
@@ -6,7 +6,17 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const stringArray = sourceString.split(';');
+  const filterArray = stringArray.filter((item) => item.trim().length);
```

Comment:
[CODE KNOWLEDGE] - Don't need to make copy before each operation. Moreover, methods like `filter` or `map` don't mutate original array. So, you don't need to create `filterArray`, you can chain `filter` method directly after `split`.